### PR TITLE
Implement filtering and reordering quest priorities

### PR DESCRIPTION
**Pull request by westnordost**
Implements #35 

For public review.

### CODE REVIEW

File: app/src/androidTest/java/de/westnordost/streetcomplete/data/visiblequests/QuestTypeOrderListTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+	public void testClear()
+	{
+		questTypeOrderList.apply(testQuestType2, testQuestType);
+		questTypeOrderList.clear();
+		questTypeOrderList.sort(list);
+
+		assertEquals(testQuestType, list.get(0));
+		assertEquals(testQuestType2, list.get(1));
+	}
+}
```
Comment (westnordost) [inline app/src/androidTest/java/de/westnordost/streetcomplete/data/visiblequests/QuestTypeOrderListTest.java]: Should probably also test:
- a QuestType is in QuestTypeOrderList but not in to-be-sorted list
- change quest type order (multiple "conflicting" apply() calls) -> should overwrite previous

File: app/src/main/java/de/westnordost/streetcomplete/location/SingleLocationRequest.java [hunk 2]
```diff
@@ -10,7 +10,7 @@
 import com.mapzen.android.lost.api.LocationServices;
 import com.mapzen.android.lost.api.LostApiClient;
 
-/** Request ONE location update */
+/** Request ONE location onBind */
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/location/SingleLocationRequest.java]: whoopsie

File: app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java [hunk 3]
```diff
@@ -128,7 +128,7 @@ else if(noProperRoad == AddRoadNameForm.IS_LINK)
 			}
 		}
 
-		// these params are passed from the form only to update the road name suggestions so that
+		// these params are passed from the form only to onBind the road name suggestions so that
```
Comment (westnordost) [inline app/src/main/java/de/westnordost/streetcomplete/quests/road_name/AddRoadName.java]: whoopsie

**Comment by matkoniecz** [comment]
> Long-press on a quest to reorder its priority for download and display on the map.

Maybe "Long-press and move a quest to reorder its priority for download and display on the map." will be more clear?

**Comment by matkoniecz** [comment]
Is it now OK to propose quests for experts that may be unfixable from SC? For example - showing fixme tags (this quest is very useful in my experience, it allowed me to fix many long-standing fixme tags)?

**Comment by matkoniecz** [comment]
Maybe I missed something but it hides wheelchair and baby changing quests without notifying current users.

Maybe give some feedback to current users (and also new ones, say after reaching 500 stars) that one may go to settings to enable additional quests?

**Comment by rugk** [comment]
> Is it now OK to propose quests for experts that may be unfixable from SC? 

That would fit to https://github.com/westnordost/StreetComplete/issues/471…

**Comment by matkoniecz** [comment]
> That would fit to https://github.com/westnordost/StreetComplete/issues/471…

You wanted to link #471

**Comment by westnordost** [comment]
> Is it now OK to propose quests for experts that may be unfixable from SC

Hm, no

**Comment by westnordost** [comment]
> Maybe I missed something but it hides wheelchair and baby changing quests without notifying current users.

I do not think this is necessary.

> Maybe give some feedback to current users (and also new ones, say after reaching 500 stars) that one may go to settings to enable additional quests?

That would be part of a larger feature to trigger certain events on reaching certain statistics. This is blocked until after we have proper user statistics.

**Comment by matkoniecz** [comment]
> This is blocked until after we have proper user statistics.

I think that for certain things, like here star count is enough (I assume that case of app reinstall resetting star count is rare enough that it is OK to repeat actions triggered by star count).

**Pull request merged**